src/sentry/preprod/api/models/project_preprod_build_details_models.py

```diff
@@ -65,24 +96,44 @@ def platform_from_artifact_type(artifact_type: PreprodArtifact.ArtifactType) ->
             raise ValueError(f"Unknown artifact type: {artifact_type}")
 
 
+def to_size_info(size_metrics: None | PreprodArtifactSizeMetrics) -> None | SizeInfo:
+    if size_metrics is None:
+        return None
+    match size_metrics.state:
+        case PreprodArtifactSizeMetrics.SizeAnalysisState.PENDING:
+            return SizeInfoPending()
+        case PreprodArtifactSizeMetrics.SizeAnalysisState.PROCESSING:
+            return SizeInfoProcessing()
+        case PreprodArtifactSizeMetrics.SizeAnalysisState.COMPLETED:
```

Comment:
One thing we haven't figured out yet is how we want to handle size metrics for non-main artifacts like watch and dynamic features. Don't have to handle here, but if we are adding a new field we might want to think about how that fits in too. I was thinking of maybe having a separate array field to handle those so that the primary use-case is easy to access, wdyt?

Reply:
Yeah that makes a lot of sense, I think with the data model we have it is sensible to have a 'main' artifact and have the non-main artifacts separate